packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -119,6 +129,39 @@ export class SCWSigner implements Signer {
         // https://eips.ethereum.org/EIPS/eip-3326#wallet_switchethereumchain
         return switched ? (null as T) : undefined;
       }
+      case SupportedEthereumMethods.WalletGetCapabilities: {
+        const params = request.params as GetCapabilitiesAction['params'];
+        const walletCapabilities = this.stateManager.walletCapabilities?.[params.chainId];
```

Comment:
we decided this should just return all chain ids capabilities. ie it does not take a chain id param, and returns an object where the keys are chain ids and values are capabilities for that chain id

Reply:
Ah okay makes sense so dapps can just check the capabilities once and cache the result, thanks for clarifying

Reply:
addressed in [f7bae2d](https://github.com/coinbase/coinbase-wallet-sdk/pull/1138/commits/f7bae2dfaa24b3d9e72c7d2ef144388b998914ed) and [da15e6c](https://github.com/coinbase/coinbase-wallet-sdk/pull/1138/commits/da15e6cdef101533d559479e8c55ea645ce0dbf0)